src/coreComponents/mesh/MeshObjectPath.cpp

```diff
@@ -0,0 +1,296 @@
+                             MeshLevel::groupStructKeys::elemManagerString() ) );
+
+  // No MeshBody or MeshLevels were specified. add all of them
+  if( objectIndex==0 )
+  {
+    pathTokens.insert( pathTokens.begin(), "{*}" );
+    pathTokens.insert( pathTokens.begin(), "{*}" );
+  }
+  // MeshBody OR MeshLevel specified. Check which one, and add all of the other.
+  else if( objectIndex==1 )
+  {
+    string const unidentifiedName = pathTokens[0];
+    // if the MeshBody is specified, add all MeshLevels
+    if( meshBodies.hasGroup( unidentifiedName ) )
+    {
+      pathTokens.insert( pathTokens.begin(), unidentifiedName );
+      pathTokens.insert( pathTokens.begin()+1, "{*}" );
+    }
+    // It wasn't the MeshBody that was specified, it was the MeshLevel?? Check and add.
+    else
+    {
+      pathTokens.insert( pathTokens.begin(), "{*}" );
+
+      string existingMeshBodyAndLevel;
+      bool levelNameFound = false;
+      meshBodies.forSubGroups< MeshBody >( [&]( MeshBody const & meshBody )
+      {
+        existingMeshBodyAndLevel += meshBody.getName() + ": ";
+        meshBody.forMeshLevels( [&]( MeshLevel const & meshLevel )
+        {
+          existingMeshBodyAndLevel += meshLevel.getName() + ", ";
+          levelNameFound = ( unidentifiedName==meshLevel.getName() ) ? true : levelNameFound;
+        } );
+        existingMeshBodyAndLevel += "/n";
+      } );
+
+      GEOSX_ERROR_IF( !levelNameFound,
+                      GEOSX_FMT( "Path ({}) specifies an invalid MeshBody or MeshLevel. ",
+                                 "existing MeshBodies: MeshLevels /n",
+                                 path,
+                                 existingMeshBodyAndLevel ) );
+      pathTokens.insert( pathTokens.begin()+1, unidentifiedName );
+    }
+  }
+
+  // checks to make sure we have properly inserted thus far
+  objectIndex = findObjectIndex();
+  size_t targetTokenLength = pathTokens.size();
+
+  GEOSX_ERROR_IF_NE_MSG( objectIndex, 2,
+                         "Filling of MeshBody and/or MeshLevel in path has failed. Object Index should be 2" );
+
+  GEOSX_ERROR_IF( targetTokenLength < 2,
+                  "Filling of MeshBody and/or MeshLevel in path has failed. targetTokenLength should be greater than 2" );
+
+  // now we need to fill in any missing region/subregion specifications.
+
+  if( m_objectType == ObjectTypes::elems )
+  {
+    // there are no regions specified
+    if( targetTokenLength == 3 )
+    {
+      pathTokens.push_back( "{*}" );
+      pathTokens.push_back( "{*}" );
+    }
+    // there are no subregions specified
+    else if( targetTokenLength == 4 )
+    {
+      pathTokens.push_back( "{*}" );
+    }
+  }
+
+
+  for( size_t a=0; a<pathTokens.size(); ++a )
+  {
+    pathTokens[a].erase( std::remove( pathTokens[a].begin(), pathTokens[a].end(), '{' ), pathTokens[a].end());
+    pathTokens[a].erase( std::remove( pathTokens[a].begin(), pathTokens[a].end(), '}' ), pathTokens[a].end());
+  }
+
+  return pathTokens;
```

Comment:
This seems correct to me but it's pretty hard to just follow like this. I guess you have tested it anyways.

Reply:
Maybe I could clear up the comments a bit...